local-interchain/interchain/types/chain_builder.go

```diff
@@ -0,0 +1,203 @@
+		Genesis: Genesis{
+			Accounts:        []GenesisAccount{},
+			Modify:          []cosmos.GenesisKV{},
+			StartupCommands: []string{},
+		},
+	}
+}
+
+func (c *Chain) SetDenom(denom string) *Chain {
+	c.Denom = denom
+	return c
+}
+
+func (c *Chain) SetDockerImage(dockerImage ibc.DockerImage) *Chain {
+	c.DockerImage = dockerImage
+	return c
+}
+
+func (c *Chain) SetHostPortOverride(hostPortOverride map[string]string) *Chain {
+	c.HostPortOverride = hostPortOverride
+	return c
+}
+
+func (c *Chain) SetGasPrices(gasPrices string) *Chain {
+	c.GasPrices = gasPrices
+	return c
+}
+
+func (c *Chain) SetGasAdjustment(gasAdjustment float64) *Chain {
+	c.GasAdjustment = gasAdjustment
+	return c
+}
+
+func (c *Chain) SetValidators(numberVals int) *Chain {
+	c.NumberVals = numberVals
+	return c
+}
+
+func (c *Chain) SetNodes(numberNode int) *Chain {
+	c.NumberNode = numberNode
+	return c
+}
+
+func (c *Chain) SetChainType(chainType string) *Chain {
+	c.ChainType = chainType
+	return c
+}
+
+func (c *Chain) SetDebugging(debugging bool) *Chain {
+	c.Debugging = debugging
+	return c
+}
+
+func (c *Chain) SetBlockTime(blockTime string) *Chain {
+	c.BlockTime = blockTime
+	return c
+}
+
+func (c *Chain) SetTrustingPeriod(trustingPeriod string) *Chain {
+	c.TrustingPeriod = trustingPeriod
+	return c
+}
+
+func (c *Chain) SetICSConsumerLink(icsConsumerLink string) *Chain {
+	c.ICSConsumerLink = icsConsumerLink
+	return c
+}
+
+// SetIBCPaths hardcodes the set IBC paths array for the chain.
+func (c *Chain) SetIBCPaths(ibcPaths []string) *Chain {
+	c.IBCPaths = ibcPaths
+	return c
+}
+
+// SetChainsIBCLink appends the new IBC path to both chains
+func (c *Chain) SetAppendedIBCPathLink(counterParty *Chain) *Chain {
+	ibcPath := fmt.Sprintf("%s_%s", c.ChainID, counterParty.ChainID)
+	c.IBCPaths = append(c.IBCPaths, ibcPath)
+	counterParty.IBCPaths = append(counterParty.IBCPaths, ibcPath)
+	return c
+}
+
+func (c *Chain) SetGenesis(genesis Genesis) *Chain {
+	c.Genesis = genesis
+	return c
+}
+
+func (c *Chain) SetDefaultSDKv47Genesis(numRandAcc int) *Chain {
+	c.Genesis = Genesis{
+		Modify: []cosmos.GenesisKV{
+			cosmos.NewGenesisKV("app_state.gov.params.voting_period", "3s"),
+			cosmos.NewGenesisKV("app_state.gov.params.max_deposit_period", "10s"),
+			cosmos.NewGenesisKV("app_state.gov.params.min_deposit.0.denom", c.Denom),
+			cosmos.NewGenesisKV("app_state.gov.params.min_deposit.0.amount", "1"),
+		},
+		Accounts: append(
+			[]GenesisAccount{
+				NewGenesisAccount("acc0", c.Bech32Prefix, "25000000000%DENOM%", c.CoinType,
+					"decorate bright ozone fork gallery riot bus exhaust worth way bone indoor calm squirrel merry zero scheme cotton until shop any excess stage laundry",
+				),
+				NewGenesisAccount("acc1", c.Bech32Prefix, "24000000000%DENOM%", c.CoinType,
+					"wealth flavor believe regret funny network recall kiss grape useless pepper cram hint member few certain unveil rather brick bargain curious require crowd raise",
+				),
+			},
```

Comment:
why do these two accounts get created here? is there a reason the genesis amounts are different?

Reply:
These are specific deterministically generated accounts found in node's test_node.sh as well. So if a user wants pre-funded accounts to be compatible with their keyring adds, this adds those in

Different amounts just to make it easier to debug & not accidentally use one when you mean to use the other (have had this happen many times, and doing this helps a ton)